### ISSUE TITLE
should be ok if the prop has initial value

### ISSUE THREAD

**Issue by hankchiutw** [hankchiutw/subjectize#1]
Setting prop's initial value is a common use case, but it invokes error with the current implementation. Specifically, make the following code work:
```js
class SomeClass {
  propA = 'A';

  @Subjectize('propA')
  propA$ = new ReplaySubject(1);
}

const instance = new SomeClass();

// should print 'A'
instance.propA$.subscribe(console.log);
```

**Comment by phoenix741** [comment]
Hello,
I have the same problem.

Thanks for the modules

**Comment by hankchiutw** [comment]
I just figured out that this is related to how the initial value is assigned.

When we `new` an instance, the properties are firstly declared without any initial value. The property decorator is also applied at this phase. After that, the initial values are assigned by order.

Currently, `@Subjectize` watches the `propA` very early and try to emit value before the `propA$` is assigned.

As a workaround, the following code would work: (yes, just change the order)
```js
class SomeClass {
  @Subjectize('propA')
  propA$ = new ReplaySubject(1);

  propA = 'A';
}

const instance = new SomeClass();

// should print 'A'
instance.propA$.subscribe(console.log);
```

**Comment by hankchiutw** [comment]
Fixed by cecccc2
